File: src/tutorial_python/scripts/publisher.py
#!/usr/bin/env python3

import rospy
from tutorial_python.msg import Person

def talker():
	rospy.init_node('talker', anonymous=True) #node name
	pub = rospy.Publisher('Person_info', Person, queue_size=10) #topic initialized of type Person	
	
	rate = rospy.Rate(2)
	
	while not rospy.is_shutdown():
		msg=Person()
		msg.first_name = "Mohamad"
		msg.last_name = "Alajouz"
		msg.age = 21
		pub.publish(msg)
		rate.sleep()

if __name__ == '__main__':
	try:
		talker()
	except rospy.ROSInterruptException:
		pass

File: src/tutorial_cpp/src/publisher.cpp
#include "ros/ros.h"
#include "tutorial_cpp/Person.h" // Correct package name

int main(int argc, char **argv)
{
  ros::init(argc, argv, "publisher_node");
  ros::NodeHandle nh;

  ros::Publisher pub = nh.advertise<tutorial_cpp::Person>("topic", 10);
  ros::Rate loop_rate(2); // Publish at 2 Hz

  while (ros::ok())
  {
    tutorial_cpp::Person msg; // Correct message type
    msg.first_name = "Mohamad";
    msg.last_name = "Alajouz";
    msg.age = 22;

    ROS_INFO("Publishing: '%s %s', age: %d", msg.first_name.c_str(), msg.last_name.c_str(), msg.age);
    pub.publish(msg);

    ros::spinOnce();
    loop_rate.sleep();
  }

  return 0;
}
